addons/stock/models/stock_warehouse.py

```diff
@@ -183,11 +183,11 @@ def write(self, vals):
                 for warehouse in self:
                     warehouse._update_partner_data(vals['partner_id'], warehouse.company_id.id)
 
-        res = super(Warehouse, self).write(vals)
-
         if vals.get('code') or vals.get('name'):
             warehouses._update_name_and_code(vals.get('name'), vals.get('code'))
 
+        res = super().write(vals)
+
```

Comment:
Explain in the commit message that the line
```python3
route.write({'name': route.name.replace(warehouse.name, new_name, 1)})
```
is called before the write. So it ends up by changing nothing since the new name is already write

Reply:
But it breaks _get_sequence_values because it expects the name and code to be already write. That's why the new parameter has been add.

Reply:
+ Stock in lower case
Task id at the very end of commit

Reply:
Thanks for the mail and the comments.
I've just updated it, I hope it's better now ?